src/Xamarin.Android.Build.Tasks/Tasks/GenerateResourceDesignerAssembly.cs

```diff
@@ -0,0 +1,352 @@
+using System.Runtime.CompilerServices;
+using System.Runtime.Versioning;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.Android.Build.Tasks;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Java.Interop.Tools.Cecil;
+using Brutal.Dev.StrongNameSigner;
+
+namespace Xamarin.Android.Tasks
+{
+	public class GenerateResourceDesignerAssembly : AndroidTask
+	{
+		internal const string DesignerAssemblyNamespace = "Microsoft.Android.Resource.Designer";
+		public override string TaskPrefix => "GRDA";
+
+		[Required]
+		public ITaskItem RTxtFile { get; set; }
+
+		public ITaskItem ResourceMap { get; set; }
+
+		[Required]
+		public bool IsApplication { get; set; }
+
+		[Required]
+		public bool DesignTimeBuild { get; set; }
+
+		[Required]
+		public ITaskItem OutputFile { get; set; }
+
+		[Required]
+		public string TargetFrameworkVersion { get; set; }
+
+		[Required]
+		public string TargetFrameworkIdentifier { get; set; }
+
+		[Required]
+		public string ProjectDir { get; set; }
+
+		[Required]
+		public ITaskItem[] Resources { get; set; }
+
+		[Required]
+		public string ResourceDirectory { get; set; }
+		public string CaseMapFile { get; set; }
+		public ITaskItem[] AdditionalResourceDirectories { get; set; }
+		public ITaskItem[] FrameworkDirectories { get; set; }
+		public bool Deterministic { get; set; }
+		public string AssemblyName { get; set; }
+		TypeReference intArray;
+		TypeReference intRef;
+		TypeReference objectRef;
+		Dictionary<string, string> resource_fixup = new Dictionary<string, string> (StringComparer.OrdinalIgnoreCase);
+
+		public override bool RunTask ()
+		{
+			using (var res = new DirectoryAssemblyResolver (this.CreateTaskLogger (), loadDebugSymbols: false)) {
+				Run(res);
+			}
+			return !Log.HasLoggedErrors;
+		}
+
+		bool Run(DirectoryAssemblyResolver res) {
+
+			foreach (var dir in FrameworkDirectories) {
+				if (Directory.Exists (dir.ItemSpec))
+					res.SearchDirectories.Add (dir.ItemSpec);
+			}
+			// ResourceDirectory may be a relative path, and
+			// we need to compare it to absolute paths
+			ResourceDirectory = Path.GetFullPath (ResourceDirectory);
+
+			string assemblyName = Path.GetFileNameWithoutExtension (OutputFile.ItemSpec);
+
+			resource_fixup = MonoAndroidHelper.LoadMapFile (BuildEngine4, CaseMapFile, StringComparer.OrdinalIgnoreCase);
+			// Generate an assembly which contains all the values in the provided
+			// R.txt file.
+			var mp = new ModuleParameters ();
+			mp.AssemblyResolver = res;
+			mp.Kind = ModuleKind.Dll;
+			var assembly = AssemblyDefinition.CreateAssembly (
+				new AssemblyNameDefinition (assemblyName, new Version (1, 0)),
+				assemblyName,
+				mp);
+
+			var module = assembly.MainModule;
+
+			module.AssemblyReferences.Clear ();
+			var netstandardAsm = AssemblyNameReference.Parse ("netstandard, Version=2.1.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51");
```

Comment:
Why a netstandard 2.1 reference instead of a .NET 8 `System.Runtime.dll` reference?  (Though I suppose netstandard won't change at all going forward, while `System.Runtime.dll` will be constantly incrementing every year, so "stability" is a fair reason…)

Reply:
mostly stability and compatability. Its a stable API.